### PR TITLE
test: test rgw multisite nightly

### PR DESCRIPTION
**Pull request by BlaineEXE**
**Description of your changes:**

In the nightly job, run the test with the latest Ceph version so we can
detect if there are RGW changes in Ceph that might break multisite. Use
a reusable GitHub action workflow to duplicate as little code as
possible.

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Comment by leseb** [comment]
Looks like #8911 has merged. @BlaineEXE any plan to resume? Thanks!

**Comment by BlaineEXE** [comment]
@subhamkrai do you want to take a look and make sure my usage of the new `setup-cluster-resources` is correct?

File: .github/workflows/rgw-multisite-test/action.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      run: |
+        tests/scripts/github-action-helper.sh deploy_second_rook_cluster
+        kubectl create -f cluster/examples/kubernetes/ceph/object-multisite-pull-realm-test.yaml
+        # wait for realms to be pulled and zone-b-multisite-store to be created
+        tests/scripts/github-action-helper.sh wait_for_rgw rook-ceph-secondary
+
+    - name: wait for ceph cluster 1 to be ready
+      shell: bash --noprofile --norc -eo pipefail -x {0}
+      run: |
+        mkdir test
```
Comment (subhamkrai) [inline .github/workflows/rgw-multisite-test/action.yml:68]: why create this dir?

File: .github/workflows/rgw-multisite-test/action.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    required: true
+  ceph-image:
+    description: Ceph image to use for the workflow (e.g., quay.io/ceph/ceph:v16.2.5)
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: setup cluster resources
+      uses: ./.github/workflows/setup-cluster-resources
```
Comment (subhamkrai) [inline .github/workflows/rgw-multisite-test/action.yml:15]: this look good to me

File: .github/workflows/rgw-multisite-test/action.yml (hunk 1 above)
Comment (BlaineEXE) [reply .github/workflows/rgw-multisite-test/action.yml:68]: I think `validate_cluster.sh` used to write data into `test/`. That's probably a carryover from then that I can remove.

**Review by subhamkrai — APPROVED** [review]
LGTM!

File: tests/scripts/github-action-helper.sh [hunk 3]
```diff
@@ -192,6 +192,16 @@ function deploy_manifest_with_local_build() {
   kubectl create -f $1
 }
 
+function replace_ceph_image() {
+  local file="$1"  # parameter 1: the file in which to replace the ceph image
+  local ceph_image="${2:-}"  # parameter 2: the new ceph image to use
+  if [[ -z ${ceph_image} ]]; then
```
Comment (leseb) [inline tests/scripts/github-action-helper.sh:198]: You don't need double brackets when calling `-z` which is an argument from the `test` command.

File: tests/scripts/github-action-helper.sh [hunk 4]
```diff
@@ -192,6 +192,16 @@ function deploy_manifest_with_local_build() {
...
+    echo "No Ceph image given. Not adjusting manifests."
+    return 0
```
Comment (leseb) [inline tests/scripts/github-action-helper.sh:200]: I believe this is implied but it doesn't hurt to explicitly have it...

**Review by leseb — APPROVED** [review]
Approved with some nits if you want to address them...

**Pull request merged**